tools/psa/build_tfm.py

```diff
@@ -160,22 +181,301 @@ def clean_up_cloned_repos():
+            msg = "Toolchain %s is not supported by %s" % (toolchain,
+                                                    TARGET_MAP[target].name)
+            raise Exception(msg)
+        tc = toolchain
+    else:
+        tc = TARGET_MAP[target].tfm_default_toolchain
+
+    delivery_dir = join(ROOT, 'targets',
+                        TARGET_MAP[target].delivery_dir)
+
+    if not os.path.exists(delivery_dir):
+        msg = "Delivery directory (delivery_dir) missing for %s" % target
+        raise FileNotFoundError(msg)
+
+    bl_sup = True if TARGET_MAP[target].tfm_bootloader_supported else False
+    return tuple([TARGET_MAP[target].name,
+                  TARGET_MAP[target].tfm_target_name,
+                  tc,
+                  delivery_dir,
+                  bl_sup])
+
+def get_mbed_supported_tfm_targets():
+    tfm_secure_targets = get_tfm_secure_targets()
+    logger.debug("Found the following TF-M targets: {}".format(
+                                                ', '.join(tfm_secure_targets)))
+
+    return [get_target_info(t) for t in tfm_secure_targets]
+
+def commit_changes(directory, target=None, toolchain=None):
+    """
+    Check for changes in `directory` and if any then commit them
+    :param directory: path to be checked for changes
+    :param target: Target name
+    :param toolchain: Toolchain name
+    """
+    # Use --intent-to-add option of git status to identify un-tracked files
+    cmd = ['git', '-C', ROOT, 'status', 'N', directory]
+    run_cmd_and_return_errorcode(cmd)
+
+    cmd = ['git', '-C', ROOT, 'diff', '--exit-code', '--quiet', directory]
+    changes_made = run_cmd_and_return_errorcode(cmd)
+
+    if changes_made:
+        if target:
+            logger.info("Change in image for %s has been detected" % target)
+        else:
+            logger.info("Change detected in directory %s" % directory)
+        cmd = ['git', '-C', ROOT, 'add', relpath(directory, ROOT)]
+        run_cmd_and_return_errorcode(cmd)
+
+        if target:
+            logger.info("Committing image for %s" % target)
+            msg = '--message="Updated secure binary for %s (%s)"' % (target,
+                                                                    toolchain)
+        else:
+            logger.info("Committing changes in directory %s" % directory)
+            msg = '--message="Updated directory %s "' % directory
+
+        cmd = ['git', '-C', ROOT, 'commit', msg]
+        run_cmd_and_return_errorcode(cmd)
+    else:
+        if target:
+            logger.info("No changes detected in %s, skipping commit" % target)
+        else:
+            logger.info("No changes detected in %s, skipping commit" %
+                                                    relpath(directory, ROOT))
+
+def run_cmake_build(configure, cmake_build_dir, bl_supported= None,
+                    tfm_target=None, tfm_toolchain=None, debug=False):
+
+    if configure:
+        cmake_cmd = ['cmake', '-GUnix Makefiles']
```

Comment:
I'd prefer to use [Ninja](https://ninja-build.org/) for building if it can be guaranteed we have [Ninja](https://ninja-build.org/) available.

`-GNinja`

Reply:
@Patater I tried `-GNinja` with Cmake and generation of build files succeeds but build fails. Looks some work is required to enable Ninja in TF-M.
```[TF-M-Builder] 15:59:59: [1/217] Building C object secure_fw/CMakeFiles/tfm_s_obj_lib.dir/core/ipc/tfm_svcalls.o.
[TF-M-Builder] 15:59:59: FAILED: secure_fw/CMakeFiles/tfm_s_obj_lib.dir/core/ipc/tfm_svcalls.o .
[TF-M-Builder] 15:59:59: /Users/devran01/Documents/devaraj/compilers/gcc/gcc-arm-none-eabi-7-2018-q2-update/bin/arm-none-eabi-gcc -DCY8C624ABZI_D44=1 -DCY_PSOC6_CM0P=1 -DDAUTH_CHIP_DEFAULT -DDOMAIN_NS=__DOMAIN_NS -DNDEBUG=1 -DTFM_CORE_DEBUG -DTFM_LEGACY_API -DTFM_LVL=1 -DTFM_MULTI_CORE_TOPOLOGY=1 -DTFM_NVCOUNTERS_ENABLE -DTFM_PSA_API -D__ARM_FEATURE_CMSE=0 -D__DOMAIN_NS=0 -D__thumb2__ -I../platform/ext/cmsis -I../platform/ext/target/cypress/psoc64 -I../platform/ext/target/cypress/psoc64/CMSIS_Driver/Config -I../platform/ext/target/cypress/psoc64/Device/Config -I../platform/ext/target/cypress/psoc64/Device/Include -I../platform/ext/target/cypress/psoc64/Native_Driver/include -I../platform/ext/target/cypress/psoc64/Native_Driver/generated_source -I../platform/ext/target/cypress/psoc64/mailbox -I../platform/ext/target/cypress/psoc64/partition -I../interface/include -I../platform/include -I../secure_fw/core/arch/include -I../secure_fw/core/ipc/include -I../ -I../secure_fw/spm -I../secure_fw/core/include -I../secure_fw/core -I../secure_fw/core/ipc -I../platform/ext/common -I../platform/ext/target/cypress/psoc64/CMSIS_Driver -I../platform/ext/driver -mcpu=cortex-m0plus  -O3 -DNDEBUG   -xc -std=c99 -fshort-enums -fshort-wchar -funsigned-char -msoft-float -ffunction-sections -fdata-sections --specs=nano.specs -Wall -Werror -Wno-format -Wno-return-type -Wno-unused-but-set-variable --depend=secure_fw/CMakeFiles/tfm_s_obj_lib.dir/core/ipc/tfm_svcalls.o.d --depend_single_line --no_depend_system_headers -o secure_fw/CMakeFiles/tfm_s_obj_lib.dir/core/ipc/tfm_svcalls.o   -c ../secure_fw/core/ipc/tfm_svcalls.c.
[TF-M-Builder] 15:59:59: arm-none-eabi-gcc: error: unrecognized command line option '--depend=secure_fw/CMakeFiles/tfm_s_obj_lib.dir/core/ipc/tfm_svcalls.o.d'.
[TF-M-Builder] 15:59:59: arm-none-eabi-gcc: error: unrecognized command line option '--depend_single_line'.
[TF-M-Builder] 15:59:59: arm-none-eabi-gcc: error: unrecognized command line option '--no_depend_system_headers'; did you mean '-Wno-system-headers'?.```

Reply:
Saddness...

Sounds like they have bugs in their CMake files.